### PR TITLE
Hide help button when not necessary

### PR DESCRIPTION
**Pull request by adguernier**
[Trello card](https://trello.com/c/kwtvF7bl/93-05-as-hayao-given-a-game-state-i-want-to-know-if-i-have-a-chance-of-winning-or-not-by-ending-a-path-on-the-next-turn)

- [X] Display which player is the connected user
- [x] hide help button if it's not user turn
- [x] hide help button at game start
- [x] hide help button at game end

### CODE REVIEW

File: webapp/public/app.css [hunk 1]
```diff
@@ -177,6 +177,15 @@ main {
   position: relative;
 }
 
+.user-info {
+  margin-bottom: 20px;
+  text-align: center;
+}
+
+.user-info h6 {
+  text-decoration: underline;
```
Comment (jonathan-marmelab) [inline webapp/public/app.css:186]: Not a good practice to underline text that is not a Link, maybe use bold or italic instead

File: webapp/views/partials/HUD.hbs [hunk 2]
```diff
@@ -1,4 +1,14 @@
 <aside class="players-hud">
+  <div class="user-info">
+    {{#if (eq game.player_black.id user.id) }}
+      <p>You are playing as</p> <h6>black</h6>
+    {{/if}}
+
+    {{#if (eq game.player_white.id user.id) }}
+      <p>You are playing as</p> <h6>white</h6>
```
Comment (jonathan-marmelab) [inline webapp/views/partials/HUD.hbs:8]: Why use `h6` here ? Headings should be used for semantic, not styling

```suggestion
      <p>You are playing as <em>white</em></p>
```

**Review by jonathan-marmelab — CHANGES_REQUESTED** [review]

File: webapp/public/app.js [hunk 3]
```diff
@@ -1,63 +1,71 @@
...
-    const allowed = tile.dataset.allowedMove;
-    if (allowed == "false") {
-      tile.classList.add("tile--allowed-move-false");
-    }
-  });
-});
-
 document.addEventListener("DOMContentLoaded", () => {
   const eventSource = new EventSource("/sse_game_resfresh");
+  console.log(eventSource);
```
Comment (jonathan-marmelab) [inline webapp/public/app.js:3]: console.log spotted

File: webapp/public/app.js [hunk 4]
```diff
@@ -1,63 +1,71 @@
...
   eventSource.onmessage = (data) => {
     window.location.reload();
   };
 });
 
-document.querySelector("#help").addEventListener("submit", (e) => {
-  e.preventDefault();
-  const messageContainer = e.target.querySelector("div");
+if (document.querySelector("#toggleHighlight")) {
+  document.querySelector("#toggleHighlight").addEventListener("click", (e) => {
```
Comment (jonathan-marmelab) [inline webapp/public/app.js:10]:
```suggestion
  document.querySelector("#toggleHighlight")?.addEventListener("click", (e) => {
```

File: webapp/public/app.js [hunk 5]
```diff
@@ -1,63 +1,71 @@
...
-      });
-
-      missingTilesForBlocked.forEach((tile) => {
-        const hexagon = document.querySelector(
-          `.tile--${tile.symbol}.tile--${tile.color}`,
-        );
-        hexagon.classList.add("tile--blink--blocked");
-      });
+if (document.querySelector("#help")) {
+  document.querySelector("#help").addEventListener("submit", (e) => {
```
Comment (jonathan-marmelab) [inline webapp/public/app.js:30]:
```suggestion
  document.querySelector("#help")?.addEventListener("submit", (e) => {
```

File: webapp/src/game/game.service.ts [hunk 6]
```diff
@@ -102,4 +102,21 @@ export class GameService {
       player.id === game.player_black.id || player.id === game.player_white.id
     );
   }
+
+  checkPlayerTurn(game: Game, player: User) {
```
Comment (jonathan-marmelab) [inline webapp/src/game/game.service.ts:106]: Not a big fan of `check` here, maybe `isPlayerTurn` is better. `check` refers to assertions imho

**Review by jonathan-marmelab — CHANGES_REQUESTED** [review]

File: webapp/src/game/game.service.ts (hunk 6 above)
Comment (adguernier) [reply webapp/src/game/game.service.ts:106]: I share your opinion.  But I wrote an another function called `isGameBelongToPlayer` and I was asked to change it to `checkGameBelongToPlayer`. So I used the same rules for this function

**Pull request merged**